File: 贪心/435无重叠区间/435.无重叠区间.java
import java.util.Arrays;
import java.util.Comparator;

/*
 * @lc app=leetcode.cn id=435 lang=java
 *
 * [435] 无重叠区间
 */

// @lc code=start
class Solution {
    public int eraseOverlapIntervals(int[][] intervals) {
        if(intervals.length == 0) 
            return 0;
        
        Arrays.sort(intervals, new Comparator<int[]>() {
            public int compare(int[] interval1, int[] interval2) {
                return interval1[1] - interval2[1];
            }
        });
        int n = intervals.length;
        int ans = 1;
        int right = intervals[0][1];
        for(int i = 1; i < n; i++) {
            if(intervals[i][0] >= right) {
                ++ans;
                right = intervals[i][1];
            }
        }

        return n-ans;
        
    }
}
// @lc code=end

File: 贪心/435无重叠区间/435.无重叠区间.js
/*
 * @lc app=leetcode.cn id=435 lang=javascript
 *
 * [435] 无重叠区间
 */

// @lc code=start
/**
 * @param {number[][]} intervals
 * @return {number}
 */
var eraseOverlapIntervals = function(intervals) {
  if(intervals.length == 0) {
      return 0
  }

  intervals.sort((a,b) => a[1] - b[1]);

  const n = intervals.length;
  let right = intervals[0][1];
  let ans = 1;
  for (let i = 1; i < n; i++) {
      if(intervals[i][0] >= right) {
          ++ans;
          right = intervals[i][1];
      }
  }

  return n-ans;
};
// @lc code=end

